### PR TITLE
[CSE 276F Submission] PlaceSphere-v1

### PR DESCRIPTION
**Pull request by astonastonaston**
The task involves grasping a sphere and placing that on top of a little bin. The task is solvable via ppo in 5000_0000 epochs.

Training command: python ppo.py --env_id="PlaceSphere-v1" --num_envs=1024 --update_epochs=8 --num_minibatches=32 --total_timesteps=50_000_000

Evaluation command: python ppo.py --env_id="PlaceSphere-v1" --evaluate --checkpoint=/path_to_final_ckpt.pt --num_eval_envs=1 --num-eval-steps=1000 --seed=2

Evaluation results:
eval_success_rate=0.8478260869565217

### CODE REVIEW

File: mani_skill/envs/tasks/tabletop/place_sphere.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        offset = pos_obj - pos_bin
+        xy_flag = (
+            torch.linalg.norm(offset[..., :2], axis=1)
+            <= 0.005
+        )
+        z_flag = torch.abs(offset[..., 2] - self.radius - self.block_half_size[0]) <= 0.005
+        is_obj_on_bin = torch.logical_and(xy_flag, z_flag)
+        is_obj_static = self.obj.is_static(lin_thresh=1e-2, ang_thresh=0.5)
+        is_obj_grasped = self.agent.is_grasping(self.obj)
+        success = is_obj_on_bin * is_obj_static * (~is_obj_grasped)
```
Comment (StoneT2000) [inline mani_skill/envs/tasks/tabletop/place_sphere.py]: the "*" can just be "&" symbols. This removes the need to do unnecessary type casting in the line below with .bool

File: mani_skill/envs/tasks/tabletop/place_sphere.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        return reward
+
+    def compute_normalized_dense_reward(self, obs: Any, action: Array, info: Dict):
+        # this should be equal to compute_dense_reward / max possible reward
+        max_reward = 13.0
+        return self.compute_dense_reward(obs=obs, action=action, info=info) / max_reward
+
+
+
+if __name__ == "__main__":
```
Comment (StoneT2000) [inline mani_skill/envs/tasks/tabletop/place_sphere.py]: remove this script here below.

**Comment by StoneT2000** [comment]
Work looks good. Just resolve conflicts with main and see my comments. Can confirm RL works as well.

**Comment by astonastonaston** [comment]
Conflicts resolved. I re-structured a bit and hopefully I didn't make their RollBall docs wrong.

**Comment by StoneT2000** [comment]
> Conflicts resolved. I re-structured a bit and hopefully I didn't make their RollBall docs wrong.

Still missing fixes for the 2 reviewed parts. see comments above

**Comment by astonastonaston** [comment]
> > Conflicts resolved. I re-structured a bit and hopefully I didn't make their RollBall docs wrong.
> 
> Still missing fixes for the 2 reviewed parts. see comments above


I can't see any comment or required change in the code... Could you specify where they are?

**Review by StoneT2000 — CHANGES_REQUESTED** [review]

**Comment by astonastonaston** [comment]
Okay now the 2 parts are fixed

**Review by StoneT2000 — APPROVED** [review]
